django_project/realtime/tasks/ash.py

```diff
@@ -36,42 +36,65 @@ def check_processing_task():
             task_status__iexact='SUCCESS'):
         task_id = ash.task_id
         result = AsyncResult(id=task_id, app=realtime_app)
-        ash.task_status = result.state
-        ash.save()
+        if result.state == 'SUCCESS':
+            Ash.objects.filter(id=ash.id).update(
+                task_status=result.state)
+            ash.refresh_from_db()
+            ash.save()
     # Checking analysis task
     for ash in Ash.objects.exclude(
             analysis_task_id__isnull=True).exclude(
             analysis_task_id__exact='').exclude(
             analysis_task_status__iexact='SUCCESS'):
         analysis_task_id = ash.analysis_task_id
         result = AsyncResult(id=analysis_task_id, app=headless_app)
-        ash.analysis_task_status = result.state
-        # Set the impact file path if success
-        if ash.analysis_task_status == 'SUCCESS':
-            ash.impact_file_path = result.result['output']['analysis_summary']
-        ash.save()
+
```

Comment:
@ismailsunni 
The reason why I did this is because sometimes result.state is SUCCESS but it actually returns status = 1 (analysis failed)

Reply:
Yeah, I found this one also. I want to add another field to store the message if it's SUCCESS but status = 1. Thanks for this.